pkg/controller/jobframework/reconciler.go

```diff
@@ -435,23 +505,47 @@ func (r *JobReconciler) stopJob(ctx context.Context, job GenericJob, object clie
 		if stoppedNow {
 			r.record.Eventf(object, corev1.EventTypeNormal, "Stopped", eventMsg)
 		}
-		return err
+		if err != nil {
+			return err
+		}
+	} else {
+		if job.IsSuspended() {
```

Comment:
This `if` is fairly surprising. Again, I suspect it might not be needed after https://github.com/kubernetes-sigs/kueue/pull/1005. Please explain if it is really needed.